router/core/transport.go

```diff
@@ -85,67 +86,80 @@ func (ct *CustomTransport) RoundTrip(req *http.Request) (*http.Response, error)
 }
 
 type TransportFactory struct {
-	preHandlers    []TransportPreHandler
-	postHandlers   []TransportPostHandler
-	retryOptions   retrytransport.RetryOptions
-	requestTimeout time.Duration
-	logger         *zap.Logger
+	preHandlers                   []TransportPreHandler
+	postHandlers                  []TransportPostHandler
+	retryOptions                  retrytransport.RetryOptions
+	requestTimeout                time.Duration
+	localhostFallbackInsideDocker bool
+	logger                        *zap.Logger
 }
 
 var _ ApiTransportFactory = TransportFactory{}
 
 type TransportOptions struct {
-	preHandlers    []TransportPreHandler
-	postHandlers   []TransportPostHandler
-	retryOptions   retrytransport.RetryOptions
-	requestTimeout time.Duration
-	logger         *zap.Logger
+	PreHandlers                   []TransportPreHandler
+	PostHandlers                  []TransportPostHandler
+	RetryOptions                  retrytransport.RetryOptions
+	RequestTimeout                time.Duration
+	LocalhostFallbackInsideDocker bool
+	Logger                        *zap.Logger
 }
 
 func NewTransport(opts *TransportOptions) *TransportFactory {
+	if opts.LocalhostFallbackInsideDocker && docker.Inside() {
```

Comment:
I'd expect this this called on config load or inside the custom transport to keep this clean here.

Reply:
I've moved it to `newServer()`, PTAL